docs/src/content/docs/reference/configuration.mdx

```diff
@@ -673,3 +673,107 @@ starlight({
 	credits: true,
 });
 ```
+
+## Configure content collections
+
+Starlight uses Astro [content collections](https://docs.astro.build/en/guides/content-collections/) to load your content.
+Starlight’s content loaders and schemas help configure collections as required.
+
+```js
+// src/content.config.ts
+import { defineCollection } from 'astro:content';
+import { docsLoader, i18nLoader } from '@astrojs/starlight/loaders';
+import { docsSchema, i18nSchema } from '@astrojs/starlight/schema';
+
+export const collections = {
+	docs: defineCollection({ loader: docsLoader(), schema: docsSchema() }),
+	// Optional: the i18n collection is used to translate UI in multilingual sites
+	i18n: defineCollection({ loader: i18nLoader(), schema: i18nSchema() }),
+};
+```
+
+### Loaders
+
+Starlight exports the following [Astro loaders](https://docs.astro.build/en/reference/content-loader-reference/) from the `@astrojs/starlight/loaders` module to simplify configuring content collections.
+
+#### `docsLoader()`
+
+The `docsLoader()` loads local Markdown, MDX, and Markdoc files from the `src/content/docs/` directory.
+File names starting with an underscore (`_`) are ignored.
+
+##### Import
+
+```js
+import { docsLoader } from '@astrojs/starlight/loaders';
+```
+
+##### Options
+
+###### `generateId()`
+
+**type:** `({ entry: string; base: URL; data: Record<string, unknown> }) => string`
+
+By default, pages generated using `docsLoader()` process your file names using a sluggifier, which removes special characters and lowercases the file name.
+If you want to override this default, provide your own custom `generateId()` function.
```

Comment:
I wonder if we should/could be helpful to provide a bit more "guidance/use-cases" here, e.g.

> If you want to override this default, e.g. to preserve the original case or some special characters in the slug, provide your own custom `generateId()` function.

Maybe it could be a hint that totally transforming the structure may not work at all for some cases/features?